### PR TITLE
[chore][receiver/mongodb] Update metric supported versions

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

Duplicate of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32779, it looks like the other can't be updated properly as it was a PR from a branch in the `contrib` repo, instead of my own fork.

### CODE REVIEW

**Comment by crobert-1** [comment]
_Adding `ready to merge` as this is a duplicate of a PR that has approval from a code owner and multiple project approvers/maintainers._

**Review by songy23 — APPROVED** [review]

**Pull request merged**
